pyccel/ast/builtin_methods/set_methods.py

```diff
@@ -200,3 +201,33 @@ def __init__(self, set_variable, item) -> None:
         if set_variable.class_type.element_type != item.class_type:
             raise TypeError("Expecting an argument of the same type as the elements of the set")
         super().__init__(set_variable, item)
+
+#==============================================================================
+class SetUpdate(SetMethod):
+    """
+    Represents a call to the .update() method.
+
+    Represents a call to the .update() method of an object with a set type,
+    Which adds items from another set (or any other iterable).
+    This method is handled through the call to `_visit_SetUpdate` in
+    the semantic stage. It then attempts to construct a `For` loop node with
+    a body that calls `add()`, or direct `add()` nodes depending on
+    the type of the iterable passed to `update()`.
+    This class should never be instantiated; it's only purpose is to help
+    construct the annotation_method `_visit_SetUpdate`. 
+    The update method is called as follows:
+
+    Parameters
+    ----------
+    set_obj : TypedAstNode
+        The set object which the method is called from.
+
+        The argument passed to update() method.
+    iterable : TypedAstNode
+        The item to search for, and remove.
```

Comment:
Is this section useful? If it is just duplicating Python docs then maybe we should just remove the unused `__init__` method?